File: ios/DailyMemoryWidget/WidgetColors.swift
import SwiftUI

/// Shared color definitions for widgets
/// Note: Widgets have limited access to main app resources, so we define colors here
enum WidgetColors {
    // Primary Brand Color
    static let primary = Color(hex: "6366F1")
    static let primaryDark = Color(hex: "4F46E5")
    static let onPrimary = Color.white

    // Surface Colors
    static let surface = Color.white
    static let background = Color(hex: "F1F5F9")

    // Text Colors
    static let textPrimary = Color(hex: "1E293B")
    static let textSecondary = Color(hex: "64748B")

    // Status Colors
    static let warning = Color(hex: "F59E0B")
    static let warningBackground = Color(hex: "FEF3C7")
    static let success = Color(hex: "22C55E")
}

// MARK: - Hex Color Extension
extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3:
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6:
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8:
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (1, 1, 1, 0)
        }

        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

File: android/app/src/main/java/com/dailymemory/presentation/widget/WidgetColors.kt
package com.dailymemory.presentation.widget

import androidx.compose.ui.graphics.Color

/**
 * Color constants for widgets
 *
 * Note: Glance widgets have limited color support compared to regular Compose.
 * These colors are used across all widget variants for consistency.
 */
object WidgetColors {
    // Primary Brand Color
    val Primary = Color(0xFF6366F1)
    val PrimaryDark = Color(0xFF4F46E5)
    val OnPrimary = Color(0xFFFFFFFF)

    // Surface Colors
    val Surface = Color(0xFFFFFFFF)
    val Background = Color(0xFFF1F5F9)

    // Text Colors
    val TextPrimary = Color(0xFF1E293B)
    val TextSecondary = Color(0xFF64748B)

    // Status Colors
    val Warning = Color(0xFFF59E0B)
    val Success = Color(0xFF22C55E)
}
